tests/e2e/xla_ops/BUILD

```diff
@@ -309,13 +307,34 @@ iree_check_single_backend_test_suite(
             "rng_uniform.mlir",  # TODO(#6995): Enable the test.
             "rng_normal.mlir",  # TODO(#6995): Enable the test.
             "round.mlir",  # TODO(#7618): Fix and enable (math.copysign -> VM)
+            "dot.mlir",  # Moved to long-timeout tests below
+            "dot_general.mlir",  # Moved to long-timeout tests below
```

Comment:
Please still consider approving this PR as-is and instead treating this as an opportunity for follow-up work:

Making tests smaller would be more work and would involve a conversation with these tests owners, and taking the decision to exclude tests from VMVX would likely also require the same conversatoin - presumably there test exist to fill a need for test coverage and some thought might already have gone into the current coverage/cost compromise.

This PR is enlarging this file, but at least it is specifically flagging the expensive tests and it provides a concrete basis for this discussion: now we know a specific reason why we care about the cost.

Reply:
I'd be okay with
* symbolic timeouts that use longer values under sanitizers
* increasing the timeout for the entire VMVX test suite
* making those tests smaller

I'm not okay with splitting the test suite in the way this PR currently has it.

Reply:
I think the simplest to allow Benoit to proceed here is just raising the timeout on vmvx tests. That's fine with me for now

Reply:
In addition to raising the timeout I also need them to have the `notsan` tag, because even though vmvx+tsan dot/dot_general tests eventually succeed, they are not worth the cost.

While raising the timeout is something that can be done on the entire suite without a concrete downside, adding the `notsan` tag to the entire suite would mean skipping all of it, not just the few expensive dot/dot_general tests, under TSan.

I'm going to do that now as it's still be simplest way forward but I wanted to flag the loss of coverage.

Reply:
Alternatively, maybe it's not so bad after all to have that conversation now about making dot/dot_general tests smaller shapes. Going to try to track down the owners...

Reply:
Conversation can continue in #9444 to decide if we can just go down the route of making these tests less expensive now.